diff --git a/cjs/utils.js b/cjs/utils.js
index 3683b2495c99882a6e5300df25985e05784a009f..969a7f915d2c492fb1928bd9c69b383040cae3f1 100644
--- a/cjs/utils.js
+++ b/cjs/utils.js
@@ -540,7 +540,7 @@ const getFieldNames = ({ selections, fieldNames = new Set(), parentName = '', lo
 exports.getFieldNames = getFieldNames;
 const getNodeComment = (node) => {
     let commentText = node.description?.value;
-    const deprecationDirective = node.directives.find(v => v.name.value === 'deprecated');
+    const deprecationDirective = node.directives?.find(v => v.name.value === 'deprecated');
     if (deprecationDirective) {
         const deprecationReason = getDeprecationReason(deprecationDirective);
         commentText = `${commentText ? `${commentText}\n` : ''}@deprecated ${deprecationReason}`;
diff --git a/esm/utils.js b/esm/utils.js
index d1e2048e98190eef033f9d272027592fdb725fb9..c7705f02525f06d3d0a00278e823ffc495fa4c67 100644
--- a/esm/utils.js
+++ b/esm/utils.js
@@ -510,7 +510,7 @@ export const getFieldNames = ({ selections, fieldNames = new Set(), parentName =
 };
 export const getNodeComment = (node) => {
     let commentText = node.description?.value;
-    const deprecationDirective = node.directives.find(v => v.name.value === 'deprecated');
+    const deprecationDirective = node.directives?.find(v => v.name.value === 'deprecated');
     if (deprecationDirective) {
         const deprecationReason = getDeprecationReason(deprecationDirective);
         commentText = `${commentText ? `${commentText}\n` : ''}@deprecated ${deprecationReason}`;
