diff --git a/language/source.js b/language/source.js
index 72ed71fdefa7d85fa12c36e67341ba6c7336d4f5..29f46205fa618e566b9045831caac97ce6129e47 100644
--- a/language/source.js
+++ b/language/source.js
@@ -4,7 +4,7 @@ exports.Source = void 0;
 exports.isSource = isSource;
 const devAssert_ts_1 = require("../jsutils/devAssert.js");
 const instanceOf_ts_1 = require("../jsutils/instanceOf.js");
-const sourceSymbol = Symbol('Source');
+const sourceSymbol = Symbol.for('graphql.Source');
 class Source {
     constructor(body, name = 'GraphQL request', locationOffset = { line: 1, column: 1 }) {
         this.__kind = sourceSymbol;
diff --git a/language/source.mjs b/language/source.mjs
index 945e9f44adf39efc27f57c095832a91190f8de27..0dd63d96dab7752c3a645c3cc3255876f2c94d9b 100644
--- a/language/source.mjs
+++ b/language/source.mjs
@@ -1,6 +1,6 @@
 import { devAssert } from "../jsutils/devAssert.mjs";
 import { instanceOf } from "../jsutils/instanceOf.mjs";
-const sourceSymbol = Symbol('Source');
+const sourceSymbol = Symbol.for('graphql.Source');
 export class Source {
     constructor(body, name = 'GraphQL request', locationOffset = { line: 1, column: 1 }) {
         this.__kind = sourceSymbol;
diff --git a/type/definition.js b/type/definition.js
index c1f4653e04cc119746a32f340692eb961685d282..98fdfc54ee449a8f54eb1c872b2a5db56e7cafec 100644
--- a/type/definition.js
+++ b/type/definition.js
@@ -80,7 +80,7 @@ function assertType(type) {
     }
     return type;
 }
-const scalarSymbol = Symbol('Scalar');
+const scalarSymbol = Symbol.for('graphql.Scalar');
 function isScalarType(type) {
     return (0, instanceOf_ts_1.instanceOf)(type, scalarSymbol, GraphQLScalarType);
 }
@@ -90,7 +90,7 @@ function assertScalarType(type) {
     }
     return type;
 }
-const objectSymbol = Symbol('Object');
+const objectSymbol = Symbol.for('graphql.Object');
 function isObjectType(type) {
     return (0, instanceOf_ts_1.instanceOf)(type, objectSymbol, GraphQLObjectType);
 }
@@ -100,7 +100,7 @@ function assertObjectType(type) {
     }
     return type;
 }
-const fieldSymbol = Symbol('Field');
+const fieldSymbol = Symbol.for('graphql.Field');
 function isField(field) {
     return (0, instanceOf_ts_1.instanceOf)(field, fieldSymbol, GraphQLField);
 }
@@ -110,7 +110,7 @@ function assertField(field) {
     }
     return field;
 }
-const argumentSymbol = Symbol('Argument');
+const argumentSymbol = Symbol.for('graphql.Argument');
 function isArgument(arg) {
     return (0, instanceOf_ts_1.instanceOf)(arg, argumentSymbol, GraphQLArgument);
 }
@@ -120,7 +120,7 @@ function assertArgument(arg) {
     }
     return arg;
 }
-const interfaceSymbol = Symbol('Interface');
+const interfaceSymbol = Symbol.for('graphql.Interface');
 function isInterfaceType(type) {
     return (0, instanceOf_ts_1.instanceOf)(type, interfaceSymbol, GraphQLInterfaceType);
 }
@@ -130,7 +130,7 @@ function assertInterfaceType(type) {
     }
     return type;
 }
-const unionSymbol = Symbol('Union');
+const unionSymbol = Symbol.for('graphql.Union');
 function isUnionType(type) {
     return (0, instanceOf_ts_1.instanceOf)(type, unionSymbol, GraphQLUnionType);
 }
@@ -140,7 +140,7 @@ function assertUnionType(type) {
     }
     return type;
 }
-const enumSymbol = Symbol('Enum');
+const enumSymbol = Symbol.for('graphql.Enum');
 function isEnumType(type) {
     return (0, instanceOf_ts_1.instanceOf)(type, enumSymbol, GraphQLEnumType);
 }
@@ -150,7 +150,7 @@ function assertEnumType(type) {
     }
     return type;
 }
-const enumValueSymbol = Symbol('EnumValue');
+const enumValueSymbol = Symbol.for('graphql.EnumValue');
 function isEnumValue(value) {
     return (0, instanceOf_ts_1.instanceOf)(value, enumValueSymbol, GraphQLEnumValue);
 }
@@ -160,7 +160,7 @@ function assertEnumValue(value) {
     }
     return value;
 }
-const inputObjectSymbol = Symbol('InputObject');
+const inputObjectSymbol = Symbol.for('graphql.InputObject');
 function isInputObjectType(type) {
     return (0, instanceOf_ts_1.instanceOf)(type, inputObjectSymbol, GraphQLInputObjectType);
 }
@@ -170,7 +170,7 @@ function assertInputObjectType(type) {
     }
     return type;
 }
-const inputFieldSymbol = Symbol('InputField');
+const inputFieldSymbol = Symbol.for('graphql.InputField');
 function isInputField(field) {
     return (0, instanceOf_ts_1.instanceOf)(field, inputFieldSymbol, GraphQLInputField);
 }
@@ -180,7 +180,7 @@ function assertInputField(field) {
     }
     return field;
 }
-const listSymbol = Symbol('List');
+const listSymbol = Symbol.for('graphql.List');
 function isListType(type) {
     return (0, instanceOf_ts_1.instanceOf)(type, listSymbol, GraphQLList);
 }
@@ -190,7 +190,7 @@ function assertListType(type) {
     }
     return type;
 }
-const nonNullSymbol = Symbol('NonNull');
+const nonNullSymbol = Symbol.for('graphql.NonNull');
 function isNonNullType(type) {
     return (0, instanceOf_ts_1.instanceOf)(type, nonNullSymbol, GraphQLNonNull);
 }
diff --git a/type/definition.mjs b/type/definition.mjs
index e488c2d718bb127532d4ff1aa8b48a5e0bcc0250..19a6ec199ff7b6dc055815ba1b99dc937784287d 100644
--- a/type/definition.mjs
+++ b/type/definition.mjs
@@ -29,7 +29,7 @@ export function assertType(type) {
     }
     return type;
 }
-const scalarSymbol = Symbol('Scalar');
+const scalarSymbol = Symbol.for('graphql.Scalar');
 export function isScalarType(type) {
     return instanceOf(type, scalarSymbol, GraphQLScalarType);
 }
@@ -39,7 +39,7 @@ export function assertScalarType(type) {
     }
     return type;
 }
-const objectSymbol = Symbol('Object');
+const objectSymbol = Symbol.for('graphql.Object');
 export function isObjectType(type) {
     return instanceOf(type, objectSymbol, GraphQLObjectType);
 }
@@ -49,7 +49,7 @@ export function assertObjectType(type) {
     }
     return type;
 }
-const fieldSymbol = Symbol('Field');
+const fieldSymbol = Symbol.for('graphql.Field');
 export function isField(field) {
     return instanceOf(field, fieldSymbol, GraphQLField);
 }
@@ -59,7 +59,7 @@ export function assertField(field) {
     }
     return field;
 }
-const argumentSymbol = Symbol('Argument');
+const argumentSymbol = Symbol.for('graphql.Argument');
 export function isArgument(arg) {
     return instanceOf(arg, argumentSymbol, GraphQLArgument);
 }
@@ -69,7 +69,7 @@ export function assertArgument(arg) {
     }
     return arg;
 }
-const interfaceSymbol = Symbol('Interface');
+const interfaceSymbol = Symbol.for('graphql.Interface');
 export function isInterfaceType(type) {
     return instanceOf(type, interfaceSymbol, GraphQLInterfaceType);
 }
@@ -79,7 +79,7 @@ export function assertInterfaceType(type) {
     }
     return type;
 }
-const unionSymbol = Symbol('Union');
+const unionSymbol = Symbol.for('graphql.Union');
 export function isUnionType(type) {
     return instanceOf(type, unionSymbol, GraphQLUnionType);
 }
@@ -89,7 +89,7 @@ export function assertUnionType(type) {
     }
     return type;
 }
-const enumSymbol = Symbol('Enum');
+const enumSymbol = Symbol.for('graphql.Enum');
 export function isEnumType(type) {
     return instanceOf(type, enumSymbol, GraphQLEnumType);
 }
@@ -99,7 +99,7 @@ export function assertEnumType(type) {
     }
     return type;
 }
-const enumValueSymbol = Symbol('EnumValue');
+const enumValueSymbol = Symbol.for('graphql.EnumValue');
 export function isEnumValue(value) {
     return instanceOf(value, enumValueSymbol, GraphQLEnumValue);
 }
@@ -109,7 +109,7 @@ export function assertEnumValue(value) {
     }
     return value;
 }
-const inputObjectSymbol = Symbol('InputObject');
+const inputObjectSymbol = Symbol.for('graphql.InputObject');
 export function isInputObjectType(type) {
     return instanceOf(type, inputObjectSymbol, GraphQLInputObjectType);
 }
@@ -119,7 +119,7 @@ export function assertInputObjectType(type) {
     }
     return type;
 }
-const inputFieldSymbol = Symbol('InputField');
+const inputFieldSymbol = Symbol.for('graphql.InputField');
 export function isInputField(field) {
     return instanceOf(field, inputFieldSymbol, GraphQLInputField);
 }
@@ -129,7 +129,7 @@ export function assertInputField(field) {
     }
     return field;
 }
-const listSymbol = Symbol('List');
+const listSymbol = Symbol.for('graphql.List');
 export function isListType(type) {
     return instanceOf(type, listSymbol, GraphQLList);
 }
@@ -139,7 +139,7 @@ export function assertListType(type) {
     }
     return type;
 }
-const nonNullSymbol = Symbol('NonNull');
+const nonNullSymbol = Symbol.for('graphql.NonNull');
 export function isNonNullType(type) {
     return instanceOf(type, nonNullSymbol, GraphQLNonNull);
 }
diff --git a/type/directives.js b/type/directives.js
index b112e92bf062153bf0ce2c8e653de808cd839303..558c610d659d1df566537910512760c0c1936a37 100644
--- a/type/directives.js
+++ b/type/directives.js
@@ -14,7 +14,7 @@ const directiveLocation_ts_1 = require("../language/directiveLocation.js");
 const assertName_ts_1 = require("./assertName.js");
 const definition_ts_1 = require("./definition.js");
 const scalars_ts_1 = require("./scalars.js");
-const directiveSymbol = Symbol('Directive');
+const directiveSymbol = Symbol.for('graphql.Directive');
 function isDirective(directive) {
     return (0, instanceOf_ts_1.instanceOf)(directive, directiveSymbol, GraphQLDirective);
 }
diff --git a/type/directives.mjs b/type/directives.mjs
index 9915b31eaca20200cb51ddb57096bc6f4ef6a6dc..76e86bb7393a2a1813f208f5898b73264992dad5 100644
--- a/type/directives.mjs
+++ b/type/directives.mjs
@@ -8,7 +8,7 @@ import { DirectiveLocation } from "../language/directiveLocation.mjs";
 import { assertName } from "./assertName.mjs";
 import { GraphQLArgument, GraphQLNonNull } from "./definition.mjs";
 import { GraphQLBoolean, GraphQLInt, GraphQLString } from "./scalars.mjs";
-const directiveSymbol = Symbol('Directive');
+const directiveSymbol = Symbol.for('graphql.Directive');
 export function isDirective(directive) {
     return instanceOf(directive, directiveSymbol, GraphQLDirective);
 }
diff --git a/type/schema.js b/type/schema.js
index 4881cba8df41accee0f581622d2de3797a3142e4..14d0382625f624f0aeb722304594b7d8045f2d49 100644
--- a/type/schema.js
+++ b/type/schema.js
@@ -19,7 +19,7 @@ function assertSchema(schema) {
     }
     return schema;
 }
-const schemaSymbol = Symbol('Schema');
+const schemaSymbol = Symbol.for('graphql.Schema');
 class GraphQLSchema {
     constructor(config) {
         this.__kind = schemaSymbol;
diff --git a/type/schema.mjs b/type/schema.mjs
index b98bf01b2a81103ee13880625a9342724ff35e18..7ea404962367f653100293d80e89eae86ffe6577 100644
--- a/type/schema.mjs
+++ b/type/schema.mjs
@@ -14,7 +14,7 @@ export function assertSchema(schema) {
     }
     return schema;
 }
-const schemaSymbol = Symbol('Schema');
+const schemaSymbol = Symbol.for('graphql.Schema');
 export class GraphQLSchema {
     constructor(config) {
         this.__kind = schemaSymbol;
